coucharchive

```diff
@@ -281,6 +281,30 @@ def replicate_couchdb_server(source_url, target_url, ignore_dbs=[]):
         pool.join()
 
 
+def create_missing_documents(source_db, target_db):
+    # Manually create documents on the target database that where previously
+    # deleted and re-created on the source. It avoids CouchDB replication bug.
+    # See: https://github.com/apache/couchdb/issues/1418
+
+    missing_documents = [doc for doc in source_db if doc not in target_db]
+
+    for doc_id in missing_documents:
+        source_doc = source_db[doc_id]
+
+        # Not needed. Will be generated on target.
+        # Plus avoids conflicts.
+        del source_doc['_rev']
+
+        try:
+            _, target_rev = target_db.save(source_doc)
+        except couchdb.http.HTTPError:
+            raise
```

Comment:
There are too many lines here :-)
What and why do you want to catch?

Please run this in a Python console:
```python
import couchdb

try:
    raise couchdb.http.HTTPError('FAKE')
except couchdb.http.HTTPError:
    raise
# what happens?

try:
    raise ValueError('FAKE 2')
except couchdb.http.HTTPError:
    raise
# what happens?

raise couchdb.http.HTTPError('FAKE')
# what happens?
```

Reply:
Ok I got it.
Since I am not going to do any operations in the `except` I don't need to `try / catch` here as an exception will be raised if an error occurs in `target_db.save()` and the program will stop.
Thanks!